src/plugins/data/server/search/strategies/rollup_search/rollup_search_strategy.test.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import { SearchStrategyDependencies } from '../../types';
+import { rollupSearchStrategyProvider } from './rollup_search_strategy';
+import { createSearchSessionsClientMock } from '../../mocks';
+
+const mockRollupResponse = {
+  body: {
+    _shards: {
+      total: 10,
+      failed: 1,
+      skipped: 2,
+      successful: 7,
+    },
+  },
+};
+
+describe('ES search strategy', () => {
+  const mockApiCaller = jest.fn();
+  const mockGetCaller = jest.fn();
+  const mockSubmitCaller = jest.fn();
+  const mockDeleteCaller = jest.fn();
+  const mockLogger: any = {
+    debug: () => {},
+  };
+  const mockDeps = ({
+    uiSettingsClient: {
+      get: jest.fn(),
+    },
+    esClient: {
+      asCurrentUser: {
+        asyncSearch: {
+          get: mockGetCaller,
+          submit: mockSubmitCaller,
+          delete: mockDeleteCaller,
+        },
+        transport: { request: mockApiCaller },
+      },
+    },
+    searchSessionsClient: createSearchSessionsClientMock(),
+  } as unknown) as SearchStrategyDependencies;
+  const mockLegacyConfig$ = new BehaviorSubject<any>({
+    elasticsearch: {
+      shardTimeout: {
+        asMilliseconds: () => {
+          return 100;
+        },
+      },
+    },
+  });
+
+  beforeEach(() => {
+    mockApiCaller.mockClear();
+    mockGetCaller.mockClear();
+    mockSubmitCaller.mockClear();
+    mockDeleteCaller.mockClear();
+  });
+
+  it('calls the rollup API', async () => {
```

Comment:
Could you please add tests with realistic error responses from the `rollup` API? (for example calling a non rollup index pattern with this API.

You can add those responses to `src/plugins/data/common/search/test_data/`

Reply:
See `src/plugins/data/server/search/strategies/ese_search/ese_search_strategy.test.ts:258`